polaris-core/src/main/java/org/apache/polaris/core/persistence/IntegrationPersistence.java

```diff
@@ -81,6 +81,26 @@ PolarisPrincipalSecrets rotatePrincipalSecrets(
       boolean reset,
       @Nonnull String oldSecretHash);
 
+  /**
+   * Reset the secrets of a principal entity, i.e. make the specified secrets as main and secondary
+   * and assign a new client id
+   *
+   * @param callCtx call context
+   * @param clientId principal client id
+   * @param principalId principal id
+   * @param customClientId the principal secret's old main secret hash
+   * @param customClientSecret the principal secret's old main secret hash
+   * @param customReset
+   */
+  @Nullable
+  PolarisPrincipalSecrets resetPrincipalSecrets(
+      @Nonnull PolarisCallContext callCtx,
+      @Nonnull String clientId,
+      long principalId,
+      String customClientId,
+      String customClientSecret,
+      boolean customReset);
```

Comment:
The value for `customReset` is always derived as `customClientId != null && customClientSecret != null`... Is the parameter really required?